flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/validation/DefaultValidator.java

```diff
@@ -385,4 +386,12 @@ private Optional<String> validateSpecChange(FlinkSessionJob sessionJob) {
 
         return Optional.empty();
     }
+
+    private Optional<String> validateServiceAccount(String serviceAccount) {
+        if (serviceAccount == null) {
+            return Optional.of(
+                    "serviceAccount must be defined. If you use helm, its value should be the same with the name of jobServiceAccount.");
```

Comment:
I think it would be better to say `spec.serviceAccount must be defined. ...` so the user knows where exactly it's missing.